flytekit/models/core/errors.py

```diff
@@ -8,7 +9,15 @@ class Kind(object):
         NON_RECOVERABLE = _errors_pb2.ContainerError.NON_RECOVERABLE
         RECOVERABLE = _errors_pb2.ContainerError.RECOVERABLE
 
-    def __init__(self, code: str, message: str, kind: int, origin: int):
+    def __init__(
+        self,
+        code: str,
+        message: str,
+        kind: int,
+        origin: int,
+        timestamp: Timestamp = Timestamp(seconds=0, nanos=0),
```

Comment:
since proto fields are all optional, should we make these optional instead?

Reply:
Proto version 3 does not seem to have the notion of optional fields. They are always default initialized. ContainerError class seemed like a wrapper around the proto, so I made it behave the same (if you construct without a timestamp or worker, you get the default values). Let me know if you have a better idea. I'll be happy to implement that.

Reply:
Can't we set to Optional/None on the python side so that in the protobuf its not set and in the backend ends up as nil? 🤔

Reply:
https://github.com/flyteorg/flytekit/blob/e19bbcc1041df1b4826a6d66cad215bf13cad4fa/flytekit/models/core/workflow.py#L117

Reply:
As a random example of a model's `to_flyte_idl`.

Reply:
@fg91, @wild-endeavor  The suggestion above is not possible with proto v3. I tested this and here is what happens. If you use None in to_flyte_idl and then get back via from_flyte_idl, it becomes Timestamp(0, 0).

Reply:
Here are examples:

```
e1 = _errors_pb2.ContainerError(worker=None)
assert e1.worker is not None
assert e1.worker == "" 
```

```
e1 = _errors_pb2.ContainerError(timestamp=None)
assert e1.timesamp is not None
assert e1.timestamp == Timestamp(0, 0) 
```

Reply:
Afaik, 'optional' keyword was brought back in a recent version, but Flyte repo has no use of it.